### ISSUE TITLE
[data grid] Grid loading overlay not covering all of the table on Windows or Linux

### ISSUE THREAD

**Issue by ducnhh8** [mui/mui-x#10871]
### Steps to reproduce

Link to live example: https://codesandbox.io/s/fervent-cookies-7xrc67?file=/src/demo.tsx

Steps:
1. Make sure you're using Edge or Chrome on Windows or Linux
2. Open the link to live example, increase the table width to see the issue more clearly, wait for data to appear
3. On the right and bottom edges you can see how the blurring is not covering all of the table content:

![Bildschirmfoto 2023-11-02 um 09 32 06](https://github.com/mui/mui-x/assets/149661492/3e56db5b-9b22-4416-8df9-43f62a5a5d45)

Another screenshot with better visual of the issue: 
![d900a1a0-69b7-464b-80b0-698290e6c5d9](https://github.com/mui/mui-x/assets/149661492/66023b06-d141-48fc-9a94-b7d23f1a6179)

⚠️ It only happens on Windows and Linux. So far I have only checked Edge and Chrome, happens to both.
Does not happen on macOS.

### Current behavior

The blurry loading overlay does not cover the entire space of the table content, when zooming in or using on large monitor screen, you can see a narrow part of the table data is not blurred by the loading overlay.

### Expected behavior

It should cover the entire table content space, at least touching the edge of the scrollbars on the left and the bottom sides.

### Context

This is the default loading overlay provided by MUI, with a small CSS adjustment for the blur:

```
componentsProps={{
  loadingOverlay: {
    sx: {
        backdropFilter: "blur(5px)"
    }
  }
}}
```

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
System:
    OS: macOS 12.5
  Binaries:
    Node: 20.2.0 - ~/.nvm/versions/node/v20.2.0/bin/node
    Yarn: Not Found
    npm: 7.24.2 - ~/Documents/react-osapiens-ui/node_modules/.bin/npm
  Browsers:
    Chrome: 118.0.5993.117
    Edge: 118.0.2088.76
    Safari: 15.6
  npmPackages:
    @emotion/react: ^11.6.0 => 11.10.0 
    @emotion/styled: ^11.6.0 => 11.10.0 
    @mui/base:  5.0.0-alpha.92 
    @mui/icons-material: ^5.2.0 => 5.8.4 
    @mui/lab: ^5.0.0-alpha.53 => 5.0.0-alpha.94 
    @mui/material: ^5.10.0 => 5.10.0 
    @mui/private-theming:  5.9.3 
    @mui/styled-engine:  5.10.0 
    @mui/styles: ^5.2.0 => 5.9.3 
    @mui/system:  5.10.0 
    @mui/types:  7.1.5 
    @mui/utils:  5.14.15 
    @mui/x-data-grid:  5.17.26 
    @mui/x-data-grid-pro: ^5.17.26 => 5.17.26 
    @mui/x-license-pro: ^5.15.0 => 5.17.12 
    @types/react: ^17.0.36 => 17.0.48 
    react: ^17.0.2 => 17.0.2 
    react-dom: ^17.0.2 => 17.0.2 
    typescript: ^4.5.2 => 4.7.4 
```
</details>

**Search keywords**: datagrid loading overlay not covering all

**Comment by DanailH** [comment]
@mui/xgrid can someone with Windows validate this?

**Comment by romgrk** [comment]
On linux I can confirm I see the same thing as the 1st screenshot. I don't think we have a windows user on the team, but I have it in dual boot, I could debug there if needed. I think we can add to the backlog but with v7 preparation it might take a bit to get done. @ducnhh8 if you want to submit a PR for this please do so, it's probably going to be faster.

**Comment by romgrk** [comment]
Actually we might want to wait until after #10059 is done, the layout will change so it might affect things.

**Comment by michelengelen** [comment]
Thanks for the report and the reproduction. This report targets v5, which reached end of life. We rebuilt the overlay system with the v7 layout rework. The overlay wrapper is now sized from the grid dimensions and covers the viewport up to the scrollbar edges (see [`GridOverlays.tsx`](https://github.com/mui/mui-x/blob/master/packages/x-data-grid/src/components/base/GridOverlays.tsx)). We also reworked the loading overlay variants in #15504 and fixed the scroll behavior in #16437. Both shipped in [v8.0.0](https://github.com/mui/mui-x/releases/tag/v8.0.0). I will close this issue as resolved in the supported versions. If you still see an uncovered strip next to the scrollbars on a current version, please open a new issue with a reproduction.
